src/apis/spawnHelpers.jsx

```diff
@@ -70,17 +63,15 @@ export const spawnForceDelete = ({ selected, path, Dialogs, setDeleteFailed, set
             });
 };
 
-export const spawnRenameItem = ({ selected, name, path, Dialogs, setErrorMessage, setHistory, setHistoryIndex }) => {
+export const spawnRenameItem = ({ selected, name, path, Dialogs, setErrorMessage }) => {
     const newPath = selected.items_cnt
         ? path.slice(0, -1).join("/") + "/" + name
         : path.join("/") + "/" + name;
 
     cockpit.spawn(renameCommand({ selected, path, name }), options)
             .then(() => {
                 if (selected.items_cnt) {
-                    cockpit.location.go("/", { path: encodeURIComponent(newPath) });
-                    setHistory(h => h.slice(0, -1));
-                    setHistoryIndex(i => i - 1);
+                    cockpit.location.replace("/", { path: encodeURIComponent(newPath) });
```

Comment:
This is wrong for two reasons:

- should check for selected.dir as well, as that is set when you use the context menu
- somehow this changes the rename behavior to make a rename switch into the renamed dir